internal/crypto/src/raw_signature/validator.rs

```diff
@@ -133,6 +133,15 @@ pub enum RawSignatureValidationError {
     /// An invalid signature value was provided.
     #[error("invalid signature value")]
     InvalidSignature,
+
+    /// The time stamp uses an unsupported signing or hash algorithm.
+    #[error("signature uses an unsupported algorithm")]
+    UnsupportedAlgorithm,
+
+    /// An unexpected internal error occured while requesting the time stamp
+    /// response.
+    #[error("internal error ({0})")]
+    InternalError(&'static str),
```

Comment:
If it's a timestamp error, why not give it a better name?

Reply:
Sounds good, thanks for clarifying (because while reading the command I thought it's specifically for a single case).